### PR TITLE
Qt4 Cleanup

### PR DESCRIPTION
**Pull request by uklotzde**
A set of trivial changes that delete all obsolete Qt4 code.

The commits have been splitted for easy review:
-  Explicit checks for any Qt 4.x version
- Checks for Qt version < 5.0, i.e. effectively also for any 4.x version
- Checks for Qt version >= 5.0, i.e. the inverse of the previous checks

Emerging merge conflicts should be trivial to resolve, e.g. #2088.

### CODE REVIEW

File: src/main.cpp
Comment (daschuer) [inline src/main.cpp]: I think this is a left over comment.

File: src/preferences/dialog/dlgprefinterface.cpp
Comment (daschuer) [inline src/preferences/dialog/dlgprefinterface.cpp]: I think we can reuse parts of that later ... mmm what does this mean here.

File: src/waveform/vsyncthread.cpp
Comment (daschuer) [inline src/waveform/vsyncthread.cpp]: This melds down to one glw->swapBuffers();
It is also part of https://github.com/mixxxdj/mixxx/pull/2088 waiting for merge. 
Can we merge this merge it first and revert this here?

File: src/waveform/vsyncthread.h
Comment (daschuer) [inline src/waveform/vsyncthread.h]: the comment above applies also here.

File: src/waveform/waveformwidgetfactory.cpp
Comment (daschuer) [inline src/waveform/waveformwidgetfactory.cpp]: conflicting with #2088

**Review by daschuer — CHANGES_REQUESTED** [review]
Thank you, I have added some comment. This this is for master and #2088 for 2.2.1 IMHO we should merge the later first. 
I don't mind to merge this to 2.2.1 as well by the way.

**Comment by uklotzde** [comment]
I will rebase this PR after merging #2088

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
Thank you for taking care. LGTM

**Pull request merged**
